### PR TITLE
implemented Mojo::mysql::Results::map and ::map_hashes

### PR DESCRIPTION
**Pull request by rndev15**

### CODE REVIEW

**Comment by jhthorsen** [comment]
What benefit does this add?

**Comment by rndev15** [comment]
Easy way to map result by key.
`$mysql->db->query('select * from posts')->map_hashes('id);`
{ 1 => { name => 'foo', id => 1}, ...}
`$mysql->db->query('select * from posts')->map;`
{ 1 => 'foo', ... }

File: lib/Mojo/mysql/Results.pm [hunk 1]
```diff
@@ -18,6 +18,24 @@ sub hash { shift->sth->fetchrow_hashref }
 
 sub hashes { Mojo::Collection->new(@{shift->sth->fetchall_arrayref({})}) }
 
+sub map {
+  return { map @$_, @{ $_[0]->arrays } };
```
Comment (jhthorsen) [inline lib/Mojo/mysql/Results.pm:22]: How does this work if you select three columns?

File: lib/Mojo/mysql/Results.pm [hunk 2]
```diff
@@ -18,6 +18,24 @@ sub hash { shift->sth->fetchrow_hashref }
...
+  my @keys;
+
+  $rows->map(sub {
+    push @keys, $_->{ $keyname };
+  });
+
+  my %return;
+  @return{@keys} = @$rows;
+  return \%return;
+}
```
Comment (jhthorsen) [inline lib/Mojo/mysql/Results.pm:37]: This is a lot of code. I would much rather run the code below, without the need of an extra method in Mojo::mysql::Results:

    %mapped = $res->hashes->map(sub { ($_->{id} => $_) })->each;

**Review by jhthorsen — COMMENTED** [review]
This PR has no documentation, so it's hard to understand what you're trying to accomplish. But I won't take it, if I have understood it correctly, since there are already quite simple ways to accomplish this.

**Comment by rndev15** [comment]
Ok, I have understand you. May be you are right, it's not necessary.
Sorry for disturb.

**Comment by jhthorsen** [comment]
No worries! Thanks for the interest 👍

**Pull request closed without merging**